spec/datetimeformat.html

```diff
@@ -239,37 +239,37 @@ <h1>Internal slots</h1>
-            <li>If the record has an [[hour]] field, it must also have a [[pattern12]] field, whose value is a String value that, in addition to the substrings of the pattern field, contains at least one of the substrings *"{ampm}"* or *"{dayPeriod}"*.</li>
-            <li>A [[rangePatterns]] field that contains a record similar to the one described in the [[formats]] field.</li>
-            <li>If the record has an [[hour]] field, it must also have a [[rangePatterns12]] field. Its value is similar to the record in [[rangePatterns]] but it uses a string similar to [[pattern12]] for each range pattern.</li>
+            <li>A [[Pattern]] field, whose value is a String value that contains for each of the date and time format component fields of the record a substring starting with *"{"*, followed by the name of the field, followed by *"}"*.</li>
+            <li>If the record has an [[Hour]] field, it must also have a [[Pattern12]] field, whose value is a String value that, in addition to the substrings of the pattern field, contains at least one of the substrings *"{ampm}"* or *"{dayPeriod}"*.</li>
+            <li>A [[RangePatterns]] field that contains a record similar to the one described in the [[Formats]] field.</li>
+            <li>If the record has an [[Hour]] field, it must also have a [[RangePatterns12]] field. Its value is similar to the record in [[RangePatterns]] but it uses a string similar to [[Pattern12]] for each range pattern.</li>
           </ol>
-          For [[DateTimeFormat]], the field value must be a string pattern which contains the strings *"{0}"* and *"{1}"*. For [[DateTimeRangeFormat]] the value of these fields must be a nested record which also has [[full]], [[long]], [[medium]] and [[short]] fields. The [[full]], [[long]], [[medium]] and [[short]] fields in the enclosing record refer to the date style of the range pattern, while the fields in the nested record refers to the time style of the range pattern. The value of these fields in the nested record is a record with a [[rangePatterns]] field and a [[rangePatterns12]] field which are similar to the [[rangePatterns]] and [rangePatterns12]] fields in [[DateFormat]] and [[TimeFormat]].
+          For [[DateTimeFormat]], the field value must be a string pattern which contains the strings *"{0}"* and *"{1}"*. For [[DateTimeRangeFormat]] the value of these fields must be a nested record which also has [[Full]], [[Long]], [[Medium]] and [[Short]] fields. The [[Full]], [[Long]], [[Medium]] and [[Short]] fields in the enclosing record refer to the date style of the range pattern, while the fields in the nested record refers to the time style of the range pattern. The value of these fields in the nested record is a record with a [[RangePatterns]] field and a [[RangePatterns12]] field which are similar to the [[RangePatterns]] and [rangePatterns12]] fields in [[DateFormat]] and [[TimeFormat]].
          </li>
```

Comment:
These [[full]]/[[long]]/etc. fields also must be lowercase to correspond with the current definition of [DateTimeStyleFormat](https://tc39.es/ecma402/#sec-date-time-style-format).
```suggestion
        <li>
          [[LocaleData]].[[&lt;_locale_&gt;]] must have a [[Styles]] field. This [[Styles]] field must be a Record with [[&lt;_calendar_&gt;]] fields for all calendar values _calendar_. The value of each [[&lt;_calendar_&gt;]] field must be a Record with all of the following fields:
          <ul>
            <li>A [[DateFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which has the same structure and constraints as a Date-Time Format Pattern Record element of the [[LocaleData]].[[<_locale_>]].[[<_calendar_>]] List.</li>
            <li>A [[TimeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which has the same structure and constraints as a Date-Time Format Pattern Record element of the [[LocaleData]].[[<_locale_>]].[[<_calendar_>]] List.</li>
            <li>A [[DateTimeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which is a string containing substrings *"{0}"* and *"{1}"* (serving as respective placeholders for time and date patterns).</li>
            <li>A [[DateTimeRangeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields (corresponding with the respective DateTimeFormat [[DateStyle]] values), the value for each of which is a Record describing a collection of format patterns for that date style. That nested Record has [[full]], [[long]], [[medium]] and [[short]] fields (corresponding with the respective DateTimeFormat [[TimeStyle]] values), the value for each of which is a Record with [[RangePatterns]] and [[RangePatterns12]] fields whose values have the same structure and constraints as the respectively-named fields of a Date-Time Format Pattern Record.</li>
          </ul>
        </li>
```